src/stirling/source_connectors/socket_tracer/protocols/mqtt/parse.cc

```diff
@@ -0,0 +1,678 @@
+                properties_length -= 2;
+                PX_ASSIGN_OR_RETURN_ERROR(std::string_view value, decoder->ExtractString((size_t)value_length));
+                properties_length -= value_length;
+                // For multiple user properties present, append to string if user property already present
+                if (result->properties.find("user-properties") == result->properties.end()) {
+                    result->properties["user-properties"] = "{" + std::string(key) + ":" + std::string(value) + "}";
+                } else {
+                    result->properties["user-properties"] += ", {" + std::string(key) + ":" + std::string(value) + "}";
+                }
+                break;
+            }
+            case static_cast<uint8_t>(PropertyCode::MaximumPacketSize): {
+                PX_ASSIGN_OR_RETURN_ERROR(uint32_t maximum_packet_size, decoder->ExtractBEInt<uint32_t>());
+                result->properties["maximum_packet_size"] = std::to_string(maximum_packet_size);
+                properties_length -= 4;
+                break;
+            }
+            case static_cast<uint8_t>(PropertyCode::WildcardSubscriptionAvailable): {
+                PX_ASSIGN_OR_RETURN_ERROR(uint8_t wildcard_subscription_available, decoder->ExtractBEInt<uint8_t>());
+                result->properties["retain_available"] = (wildcard_subscription_available == 1)?"true":"false";
+                properties_length -= 1;
+                break;
+            }
+            case static_cast<uint8_t>(PropertyCode::SubscriptionIdentifiersAvailable): {
+                PX_ASSIGN_OR_RETURN_ERROR(uint8_t subscription_id_available, decoder->ExtractBEInt<uint8_t>());
+                result->properties["subscription_id_available"] = (subscription_id_available == 1)?"true":"false";
+                properties_length -= 1;
+                break;
+            }
+            case static_cast<uint8_t>(PropertyCode::SharedSubscriptionAvailable): {
+                PX_ASSIGN_OR_RETURN_ERROR(uint8_t shared_subscription_available, decoder->ExtractBEInt<uint8_t>());
+                result->properties["subscription_id_available"] = (shared_subscription_available == 1)?"true":"false";
+                properties_length -= 1;
+                break;
+            }
+            default:
+                return ParseState::kInvalid;
+        }
+    }
+    return ParseState::kSuccess;
+}
+
+ParseState ParseVariableHeader(Message* result, BinaryDecoder* decoder, MqttControlPacketType& control_packet_type) {
+    switch (control_packet_type) {
+        case MqttControlPacketType::CONNECT: {
+            PX_ASSIGN_OR_RETURN_ERROR(uint8_t protocol_name_length, decoder->ExtractBEInt<uint16_t>());
+            PX_ASSIGN_OR_RETURN_ERROR(std::string_view protocol_name,decoder->ExtractString((size_t)protocol_name_length));
+            CTX_DCHECK(protocol_name == "MQTT");
```

Comment:
Is MQIsdp used for MQTTv3? I am not sure if the implementation will work for mqttv3 since mqtt v3 does not have properties. There might be other v5 specific code that might not work for v3 since I have not thoroughly gone through v3. I need to look more into this.

Reply:
When I made this comment it was before I read the MQTT v5 docs (I had trouble finding it first so I went with v3). It does seem like that specifically mentions `"MQTT"`, so let's just drop this.

I agree that we would need some changes to support MQTT v3. I think it's fine to target the latest version and not increase the scope of this initial implementation.